jobsdb/jobsdb.go

```diff
@@ -1170,17 +1168,18 @@ func (jd *HandleT) checkIfMigrateDS(ds dataSetT) (bool, int) {
 		return false, 0
 	}
 
-	// If records are newer than what is required. One example use case is
-	// gateway DB where records are kept to dedup
-
-	var lastUpdate time.Time
-	sqlStatement = fmt.Sprintf(`SELECT MAX(created_at) from "%s"`, ds.JobTable)
+	var minCreatedAt, maxCreatedAt time.Time
+	sqlStatement = fmt.Sprintf(`SELECT MIN(created_at), MAX(created_at) from "%s"`, ds.JobTable)
 	row = jd.dbHandle.QueryRow(sqlStatement)
-	err = row.Scan(&lastUpdate)
+	err = row.Scan(&minCreatedAt, &maxCreatedAt)
 	jd.assertError(err)
-	if jd.dsRetentionPeriod > time.Duration(0) && time.Since(lastUpdate) < jd.dsRetentionPeriod {
+
+	if jd.minDSRetentionPeriod > time.Duration(0) && time.Since(maxCreatedAt) < jd.minDSRetentionPeriod {
```

Comment:
Why do we need these checks? Just to protect ourselves from a bad input?
```
jd.minDSRetentionPeriod > time.Duration(0)
jd.maxDSRetentionPeriod > time.Duration(0)
```

Reply:
Yes